src/switchroot/ostree-remount.c

```diff
@@ -119,6 +159,49 @@ main (int argc, char *argv[])
   if (mount ("none", "/sysroot", NULL, MS_REC | MS_PRIVATE, NULL) < 0)
     perror ("warning: While remounting /sysroot MS_PRIVATE");
 
+  const char *transient_etc = NULL;
+  g_variant_dict_lookup (ostree_run_metadata, OTCORE_RUN_BOOTED_KEY_TRANSIENT_ETC, "&s",
+                         &transient_etc);
+
+  if (transient_etc)
+    {
+      /* Systemd will create a /run/machine-id -> /etc/machine-id bind mount if /etc is
```

Comment:
Historically `ostree-remount.service` has just been a big source of race conditions.

In general the direction has been to ensure that the target `/` is set up fully in the initramfs.


In particular the historical rationale for `ostree-remount.service` was compatibility with the legacy behavior of remounting `/` as writable only in the real root early boot.

If that's done in the initramfs, there is no reason to run `ostree-remount.service`.  And I think we should aim to do that uniformly - and actually disable `ostree-remount.service` by default.

I'd need to dig in here into what point the `/etc/machine-id` is being dealt with in systemd, but it really seems to me like the relevant code here should be run in the initramfs.

Reply:
> If that's done in the initramfs, there is no reason to run ostree-remount.service. 

Ah wait, there is one load bearing thing left that it's doing, which is to reset the propagation mode of `/sysroot`.

Blah...I wonder if we can work with systemd upstream to add some way to opt out of that for particular mounts.

Alternatively...I could imagine actually hiding the `/sysroot` mount in a separate mount namespace entirely, held open by some process that we keep running from the initramfs.

That would be much stronger isolation, but it'd mean every tool that wants to operate on that state would need to `nsenter -m -t $(pidof ostree-mountns-holder)`.

Reply:
It is impossible to do this relabeling in the initrd, becase at that point the selinux policy is not loaded.

Reply:
And, for reasons described in https://github.com/ostreedev/ostree/pull/2972#issuecomment-1674558770 it isn't even possible to manually fake the relabeling in the initrd. It has to be done after the policy is loaded.